Refactor BaseStrategyV1 and related contracts to replace Azorius module with Proposal Initializer

adamgall opened the pull request:
- Updated BaseStrategyV1 to use a proposalInitializer instead of azoriusModule for authorization.
- Removed the setAzorius function and associated event, replacing it with a new initialize function that accepts a proposalInitializer.
- Modified LinearERC20VotingV1 and LinearERC721VotingV1 to align with the new proposalInitializer logic.
- Updated tests to reflect changes in initialization and authorization checks.

These changes enhance the clarity and functionality of the strategy contracts, ensuring proper access control for proposal initialization.

dacebt commented on contracts/deployables/strategies/LinearERC20VotingV1.sol:
```diff
@@ -117,7 +117,7 @@ contract LinearERC20VotingV1 is
         governanceToken = IVotes(_governanceToken);
         __Ownable_init(_owner);
         __UUPSUpgradeable_init();
-        _setAzorius(_azoriusModule);
```
Something doesn't feels off by this. requiring each strategy to have this line of code for something that is required for the contract to function properly. 

I guess my question comes down to, is there value in those "non 0" 
checks prior to setting the azorius module? if yes, then we should probably keep the internal method and reuse that? 

if no, should we lock all of our strategies to this pattern? Would we ever want something else to authorize starting a proposal?

adamgall replied:
Addressed in latest (force pushed) commit

adamgall committed:
Refactor BaseStrategyV1 and related contracts to replace Azorius module with Proposal Initializer

- Updated BaseStrategyV1 to use a proposalInitializer instead of azoriusModule for authorization.
- Removed the setAzorius function and associated event, replacing it with a new initialize function that accepts a proposalInitializer.
- Modified LinearERC20VotingV1 and LinearERC721VotingV1 to align with the new proposalInitializer logic.
- Updated tests to reflect changes in initialization and authorization checks.

These changes enhance the clarity and functionality of the strategy contracts, ensuring proper access control for proposal initialization.
